tests/test_train.py

```diff
@@ -1,3 +1,106 @@
 import pytest,fastai
+from utils.fakes import *
+from utils.text import *
+from utils.param import *
+#from fastai.vision import *
 
-#I'm all empty, please fill me with tests about the functions in train.py
\ No newline at end of file
+## filename: test_train.py
+## tests functions in train.py
+
+## Class Learner, see documentation:  https://docs.fast.ai/train.html
+ 
+## Model fitting methods
+
+## run: pytest tests/test_train.py (add -s for screenoutput )
+
+
+@pytest.fixture(scope="module")
+def learn():
+    ## fixture for same model in fit and fit_one_cycle and demonstrate difference in behavior
+    ## test_fit and test_fit_one_cycle assert correctly with decommented models, 
+    ## fake_learner defaulted for performance reasons
+    
+    learn = fake_learner(50,50)
+
+    #path = untar_data(URLs.MNIST_TINY)
+    #data = ImageDataBunch.from_folder(path, ds_tfms=(rand_pad(2, 28), []), num_workers=2)
+    #data.normalize()
+    #learn = create_cnn(data, models.resnet34, metrics=error_rate)
+    #learn = Learner(data, simple_cnn((3,16,16,16,2), bn=True), metrics=[accuracy, error_rate])        
+    
+    return learn
+
+def test_fit(learn):
+
+    ## Test confirms learning rate and momentum are stable, see difference to test_fit_one_cycle
+    
+    learning_rate = 3e-3
+    weight_decay = 1e-2
+    eps = 4
+    
+    with CaptureStdout() as cs:  learn.fit(epochs=eps, lr=learning_rate, wd=weight_decay)
+    #assert_screenout(cs.out, str(eps))
+    lrs = get_learning_rates(learn)
```

Comment:
This is awfully long to just test if learn.recorder.lrs only has one element: `set(learn.recorder.lrs)` would be quicker, don't you think?

Reply:
Done (with list(...)). Had grabbed the pattern from test_callbacks_csv_logger.get_train_losses. But its s different context, so indeed should be shorter